src/main/java/com/dnastack/wes/api/WesV1Controller.java

```diff
@@ -123,36 +126,68 @@ public RunId cancelRun(@PathVariable("runId") String runId) {
     }
 
     @AuditActionUri("wes:run:stdout")
     @PreAuthorize("@accessEvaluator.canAccessResource('/ga4gh/wes/v1/runs/' + #runId, 'wes:runs:read', 'wes')")
     @GetMapping(value = "/runs/{runId}/logs/task/{taskId}/stdout", produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
-    public void getStdout(HttpServletResponse response, @PathVariable String runId, @PathVariable String taskId) throws IOException {
-        adapter.getLogBytes(response.getOutputStream(), runId, taskId, "stdout");
+    public void getTaskStdout(
+        HttpServletResponse response,
+        @RequestHeader HttpHeaders headers,
+        @PathVariable String runId,
+        @PathVariable String taskId
+    ) throws IOException {
+        adapter.getLogBytes(response.getOutputStream(), runId, taskId, "stdout",getRangeFromHeaders(headers));
     }
 
     @AuditActionUri("wes:run:stderr")
     @PreAuthorize("@accessEvaluator.canAccessResource('/ga4gh/wes/v1/runs/' + #runId, 'wes:runs:read', 'wes')")
     @GetMapping(value = "/runs/{runId}/logs/task/{taskName}/{index}/stderr", produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
-    public void getStderr(HttpServletResponse response, @PathVariable String runId, @PathVariable String taskName, @PathVariable int index) throws IOException {
-        adapter.getLogBytes(response.getOutputStream(), runId, taskName, index, "stderr");
+    public void getTaskStderr(
+        HttpServletResponse response,
+        @RequestHeader HttpHeaders headers,
+        @PathVariable String runId,
+        @PathVariable String taskName,
+        @PathVariable int index
+    ) throws IOException {
+        adapter.getLogBytes(response.getOutputStream(), runId, taskName, index, "stderr",getRangeFromHeaders(headers));
     }
 
     @AuditActionUri("wes:run:stdout")
     @PreAuthorize("@accessEvaluator.canAccessResource('/ga4gh/wes/v1/runs/' + #runId, 'wes:runs:read', 'wes')")
     @GetMapping(value = "/runs/{runId}/logs/task/{taskName}/{index}/stdout", produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
-    public void getStdout(HttpServletResponse response, @PathVariable String runId, @PathVariable String taskName, @PathVariable int index) throws IOException {
-        adapter.getLogBytes(response.getOutputStream(), runId, taskName, index, "stdout");
+    public void getTaskStdout(
+        HttpServletResponse response,
+        @RequestHeader HttpHeaders headers,
+        @PathVariable String runId,
+        @PathVariable String taskName,
+        @PathVariable int index
+    ) throws IOException {
+        adapter.getLogBytes(response.getOutputStream(), runId, taskName, index, "stdout", getRangeFromHeaders(headers));
+    }
+
+    private HttpRange getRangeFromHeaders(HttpHeaders headers){
+        List<HttpRange> ranges = headers.getRange();
+        if (ranges.isEmpty()){
+            return null;
+        } else {
+            // only return the first range parsed
+            return ranges.get(0);
```

Comment:
I wonder how strict we want to be. We can also return `416 Range Not Satisfiable` with some additional info in the error message. It might be a little bit more confusing for a client to understand why my other ranges were ignored.